src/preload.py

```diff
@@ -427,15 +427,45 @@ def list_images(detect_flasher_type_images):
     return _list_images()
 
 
+def _is_non_empty_folder(folder):
+    # True if the folder has at least one non hidden file
+    if not os.path.exists(folder) or not os.path.isdir(folder):
+        return False
+    return any(f for f in os.listdir(folder) if not x.startswith("."))
+
+
+def _find_docker_aufs_root(mountpoint):
+    # We're looking for a /docker/aufs/diff/<xxxxxxxxxxxxx>/ folder with some
+    # files not starting with a '.'
+    path = os.path.join(mountpoint, "docker", "aufs", "diff")
+    if os.path.exists(path) and os.path.isdir(path):
+        for folder in os.listdir(path):
+            folder_path = os.path.join(path, folder)
+            if _is_non_empty_folder(folder_path):
+                return folder_path
+
+
+def _find_docker_overlay2_root(mountpoint):
+    # We're looking for a /docker/overlay2/<xxxxxxxxxxxxx>/diff folder with
+    # some files not starting with a '.'
+    path = os.path.join(mountpoint, "docker", "overlay2")
+    if os.path.exists(path) and os.path.isdir(path):
+        for folder in os.listdir(path):
+            folder_path = os.path.join(path, folder, "diff")
+            if _is_non_empty_folder(folder_path):
+                return folder_path
```

Comment:
You can replace these with:
```python
# You probably want a better function name
def find_non_empty_path(path, childDir = ''):
    if os.path.exists(path) and os.path.isdir(path):
        for folder in os.listdir(path):
            folder_path = os.path.join(path, folder, childDir) if childDir != '' else os.path.join(path, folder)
            if _is_non_empty_folder(folder_path):
                return folder_path

def _find_docker_aufs_root(mountpoint):
    path = os.path.join(mountpoint, 'docker', 'aufs', 'diff')
    return find_non_empty_path(path)

def _find_docker_overlay2_root(mountpoint):
    path = os.path.join(mountpoint, 'docker', 'overlay2')
    return find_non_empty_path(path, 'diff')
```
It's untested but it cleans up this code block a bit.